doc/guide/dynamics/dynamics-krylov.rst

```diff
@@ -0,0 +1,96 @@
+
+With the above approximation, the time-evolution is calculated only with a smaller square matrix of the desired size. Therefore, the Krylov method provides huge speed-ups in computation of short-time evolutions when the dimension of the Hamiltonian is very large, a point at which exact calculations on the complete subspace are practically impossible. 
+
+One of the biggest problems with this type of method is the control of the error. After a short time, the error starts to grow exponentially. However, this can be easily corrected by restarting the subspace when the error reaches a certain threshold. Therefore, a series of :math:`M` Krylov-subspace time evolutions provides accurate solutions for the complete time evolution. Within this scheme, the magic of Krylov resides not only in its ability to capture complex time evolutions from very large Hilbert spaces with very small dimenions :math:`M`, but also in the computing speed-up it presents.
+
+For exceptional cases, the Lanczos algorithm might arrive at the exact evolution of the initial state at a dimension :math:`M_{hb}<M`. This is called a happy breakdown. For example, if a Hamiltonian has a symmetry subspace :math:`D_{\text{sim}}<M`, then the algorithm will optimize using the value math:`M_{hb}<M`, at which the evolution is not only exact but also cheap.
+
+.. _krylov-qutip:
+
+Krylov Solver in QuTiP
+======================
+
+In QuTiP, Krylov-subspace evolution is implemented as the function :func:`qutip.krylovsolve`. Arguments are nearly the same as :func:`qutip.mesolve`
+function for master-equation evolution, except that the initial state must be a ket vector, as opposed to a density matrix, and the additional parameter ``krylov_dim`` that defines the maximum allowed Krylov-subspace dimension. The maximum number of allowed Lanczos partitions can also be determined using the :func:`qutip.solver.options.nsteps` parameter, which defaults to '10000'.
+
+Let's solve a simple example using the algorithm in QuTiP to get familiar with the method.
+
+.. plot::
+    :context:
+
+    dim = 100
+    e_ops = [jmat((dim - 1) / 2.0, "x"), jmat((dim - 1) / 2.0, "y"), jmat((dim - 1) / 2.0, "z")]
+    H = .5*jmat((dim - 1) / 2.0, "z") + .5*jmat((dim - 1) / 2.0, "x")
+    psi0 = rand_ket(dim)
+    tlist = np.linspace(0.0, 10.0, 200)
+    results = krylovsolve(H, psi0, tlist, krylov_dim=20, e_ops=e_ops)
+    plt.figure()
+    for expect in results.expect:
+        plt.plot(tlist, expect)
+    plt.legend(('jmat x', 'jmat y', 'jmat z'))
+    plt.xlabel('Time')
+    plt.ylabel('Expectation values')
+    plt.show()
+
+.. _krylov-sparse:
+
+Sparse and Dense Hamiltonians
+-----------------------------------
+
+If the Hamiltonian of interest is known to be sparse, :func:`qutip.krylovsolve` also comes equipped with the possibility to store its internal data in a sparse optimized format using scipy. This allows for significant speed-ups, let's showcase it:
+
+
+.. plot::
+    :context:
```

Comment:
```suggestion
    :context:
    :nofigs:
```
Currently this block re-displays the previous plot. Adding `:nofigs:` will prevent that.